tests/unit/amo/components/TestAddonReviewList.js

```diff
@@ -341,6 +368,32 @@ describe(__filename, () => {
       sinon.assert.calledWith(dispatch, setViewContext(addon.type));
     });
 
+    it('does not dispatch a view context if there is no add-on', () => {
+      const dispatch = sinon.stub(store, 'dispatch');
+      const errorHandler = createStubErrorHandler();
+      render({ errorHandler });
+
+      // Note: Because we expect setViewContext NOT to be called, we don't have a specific
+      // addon.type argument to use for an assert.neverCalledWith, so instead we verify that
```

Comment:
Can you use `sinon.assert.notCalledWithMatch(dispatch, sinon.match({ type: '...' }))`? I haven't tried it.